packages/expo/android/src/main/java/expo/modules/ExpoReactHostFactory.kt

```diff
@@ -41,6 +41,12 @@ object ExpoReactHostFactory {
     override val jsBundleLoader: JSBundleLoader
       get() {
         val context = weakContext.get() ?: throw IllegalStateException("Unable to get concrete Context")
+        reactNativeHostWrapper.jsBundleFile?.let { jsBundleFile ->
```

Comment:
Would it be helpful to add comments saying which parts of the code run in bridgeless mode vs. with the bridge enabled? Later, when there is only bridgeless mode, we can remove the comments.

Reply:
good question! basically this whole ExpoReactHostFactory file is for bridgeless. react-native has a convention
 - bridgeless: `ReactHost`, `ReactInstance`
 - bridge: `ReactNativeHost`, `ReactInstanceMananger`, `Catalyst*`

there's some effort to adapt from ReactNativeHost to ReactHost without much change. in this example, it [reads some necessary properties from ReactNativeHost](https://github.com/facebook/react-native/blob/ad9f0b0ab88218beb214859e0d85d50f9caa7bb5/packages/react-native/template/android/app/src/main/java/com/helloworld/MainApplication.kt#L32-L33).

basically i would try to follow the same effort to migrate our [ReactNativeHostHandler](https://github.com/expo/expo/blob/main/packages/expo-modules-core/android/src/main/java/expo/modules/core/interfaces/ReactNativeHostHandler.java) to be bridgeless/bridge independent, so that expo modules don't have to handle these different cases. though expo-dev-client and expo-updates are two special cases that they are some tight coupled code inside. we are aware and working on the migration.

Reply:
Ah okay. Thanks for the explanation. If I understand correctly, does that mean that when bridged mode is removed (one day, in the future), we'll move the code from ReactNativeHostWrapper into ReactHostFactory, since ReactNativeHost will no longer exist?

Reply:
> If I understand correctly, does that mean that when bridged mode is removed (one day, in the future), we'll move the code from ReactNativeHostWrapper into ReactHostFactory, since ReactNativeHost will no longer exist?

right! we will merge into one class. the class name might be ReactHostWrapper, depending on how MainApplication from the template changes over time. if the [logic for ReactNativeHost](https://github.com/expo/expo/blob/3d13e80d72893f7429a237a47ed74be4c5388eb6/templates/expo-template-bare-minimum/android/app/src/main/java/com/helloworld/MainApplication.kt#L20-L36) moved to ReactHost, then we may have a ReactHostWrapper after all.